### PR TITLE
QUnit.diff cleanup

### PR DESCRIPTION
**Pull request by leobalter**
Trying to cleanup QUnit.diff, goal is to make it easier to maintain.

### CODE REVIEW

File: test/reporter-html/diff.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
+QUnit.module( "diff" );
```
Comment (jzaefferer) [inline test/reporter-html/diff.js]: Why is this in the `reporter-html` folder? The relevant code isn't attached to the reporter.

File: test/reporter-html/diff.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+QUnit.test( "throws if arguments are not strings", function( assert ) {
+	assert.throws(function() { QUnit.diff( {}, "" ); });
+	assert.throws(function() { QUnit.diff( "", {} ); });
```
Comment (jzaefferer) [inline test/reporter-html/diff.js]: I think these two are enough to verify that there is some argument checking in place, no need for further tests.

File: test/reporter-html/diff.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+});
+
+QUnit.test( "different strings", function( assert ) {
+	var a = "abcd";
+	var b = "xkcd";
+
+	assert.equal(
+		QUnit.diff( a, b ),
+		"<del>ab</del><ins>xk</ins><span>cd</span>",
+		"QUnit.diff( a, b )"
```
Comment (jzaefferer) [inline test/reporter-html/diff.js]: This doesn't seem like a particularly useful message. Also applies to the others.

**Comment by jzaefferer** [comment]
I've reviewed all commits individually (the only sane option here, I think). A few minor notes, looking very good overall.

**Comment by leobalter** [comment]
Thanks, I'll fix it when I get at home

File: test/reporter-html/diff.js (hunk 1 above)
Comment (leobalter) [inline test/reporter-html/diff.js]: Our current diff is only relevant to the HTML reporter as its output is HTML. These tests intend to progress the coverage on the diff functions, but they are meant for what's expected on the reporter, meaning we could change it for any other library in a future.

File: test/reporter-html/diff.js (hunk 2 above)
Comment (leobalter) [inline test/reporter-html/diff.js]: A practice I got after test262 where all cases were relevant, but that's about a language test. :)

**Comment by leobalter** [comment]
Waiting feedback to decide on https://github.com/jquery/qunit/pull/850#discussion-diff-37100759 but everything is fixed

File: test/reporter-html/diff.js (hunk 1 above)
Comment (jzaefferer) [inline test/reporter-html/diff.js]: Okay, we can leave it here.

**Comment by jzaefferer** [comment]
Good to land.

**Pull request closed without merging**